go-controller/pkg/libovsdbops/db_index.go

```diff
@@ -0,0 +1,102 @@
+package libovsdbops
+
+import (
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+)
+
+const (
+	// db ownership ExternalIDs
+	// when possible, use DbIndex and function from this file instead of directly using the following ExternalIDs
+	OwnerController = "owner_controller"
+	OwnerObjectType = "owner_object_type"
+	OwnerNamespace  = "owner_namespace"
+	OwnerName       = "owner_name"
+	InternalID      = "internal_id"
+
+	// Owner Object Types
+	EgressFirewallDNSOwnerType = "EgressFirewallDNS"
+	EgressQoSOwnerType         = "EgressQoS"
+	NetworkPolicyOwnerType     = "NetworkPolicy"
+	NamespaceOwnerType         = "Namespace"
+	HybridNodeRouteOwnerType   = "HybridNodeRoute"
+)
+
+type DbIndex struct {
+	// OwnerController specifies which controller owns the object
+	// Controller should only change objects it owns, make sure to always set this field when searching with predicate
+	OwnerController string
+	// OwnerObjectType should identify object type that owns db object, e.g. Pod, NetworkPolicy, etc.
+	// Controller handlers should use unique OwnerTypes to only change objects they manage.
+	OwnerObjectType string
+	// OwnerNamespace and OwnerName should provide unique index for objects owned by the same OwnerObjectType,
+	// e.g. if there is only 1 object owned by OwnerObjectType in the system, both Namespace and Name may be empty.
+	OwnerNamespace string
+	OwnerName      string
+	// InternalID may be used to additionally identify db objects owned by same OwnerNamespace and OwnerName.
+	InternalID string
+}
+
+func GetDbIndexFromExternalIDs(externalIDs map[string]string) *DbIndex {
+	return &DbIndex{
+		externalIDs[OwnerController],
+		externalIDs[OwnerObjectType],
+		externalIDs[OwnerNamespace],
+		externalIDs[OwnerName],
+		externalIDs[InternalID],
+	}
+}
+
+func (dbIndex *DbIndex) GetExternalIDs() map[string]string {
+	externalIDs := map[string]string{
+		OwnerController: dbIndex.OwnerController,
+		OwnerObjectType: dbIndex.OwnerObjectType,
```

Comment:
Can `OwnerObjectType` be empty? Other parts of the file do give the impresion that it can, but not here.

Reply:
Good point, it actually shouldn't be empty, will update